### DISCUSSION TITLE
Question: How to run all tests in 1 runner?

### DISCUSSION THREAD

**Discussion by moglistree** [lyndsey-ferguson/fastlane-plugin-test_center#345] [Q&A] [answered]
Hello,
Our project structure is in the following way. We have a top lvl target that is dependent on series of feature modules that are then dependent on base modules like networking etc. When using `multi_scan` on the top lvl target that in its test config has all of its dependencies tests configured to run in one go we get one runner per dependency. If we do the same config on `scan` we get all of the test ran sequentially.
Is it possible to run all the tests sequentially like on the `scan` run?
If yes how do we do this?
If no what suggestion do others have for similar situations?

**Comment by lyndsey-ferguson** [comment, marked as answer]
Are you saying that you have tests that set up test state for subsequent tests? Let me confirm that before I keep tying trying to guess your setup.

**Reply by moglistree** [reply]
We do not have tests that set up state for subsequent tests.
We have tests grouped in multiple schemes for every submodule in our app. They in turn are all triggered to run as a group for the main app target and this works with using `scan`. When using `multi_scan` the tests for every sub-module are ran by a separate runner and their results and coverage are reported separately in contrast to `scan` where all of this is reported as one. Is it possible to replicate the scan behaviour with `multi_scan` ?

**Reply by lyndsey-ferguson** [reply]
When you write `sub-module`, is that the same as a Target (I'm old school Xcode)? If so, no, you won't be able to have them run all together as the test result reports are different for each target.

**Reply by moglistree** [reply]
When I say `sub-module` I mean projects in the same workspace as the main project that might have a separate compilation target but is not required. The main thing is that these projects output a `.framework` files that are dependencies without that the main project can not compile. So their tests are ran as a total tests suite for the application. Seems like from your answer that the runner can not run them all together as the `scan` by itself can do it. A confirmation from your side would be gr8.

**Reply by lyndsey-ferguson** [reply]
Ok, so, each framework is compiled separately, right? From what I remember, that would require a separate target. Do you really care if the tests are run separately, or you are more interested in getting a complete report at the end?

**Reply by moglistree** [reply]
The important part is to get the complete report at the end.

**Reply by lyndsey-ferguson** [reply]
Ok, use `:collate_reports` option and it _should_ collate all the separate reports into one.

_([here](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/docs/feature_details/multi_scan.md#parameters) are the other options)_

**Reply by lyndsey-ferguson** [reply]
@moglistree did my answer help you?

**Reply by moglistree** [reply]
It does seem to work for us. Appreciate the time you spent helping.
